src/cli/main.zig

```diff
@@ -897,9 +939,43 @@ fn rocRun(allocs: *Allocators, args: cli_args.RunArgs) !void {
         return err;
     };
 
-    // Use native detection (typically musl) for shim generation to match embedded shim library
+    // Use native detection for shim generation to match embedded shim library
     const shim_target = builder.RocTarget.detectNative();
 
+    // Validate platform header has targets section and supports native target
+    if (platform_paths.platform_source_path) |platform_source| {
+        if (platform_validation.validatePlatformHeader(allocs.arena, platform_source)) |validation| {
+            // Validate that the native target is supported
+            platform_validation.validateTargetSupported(validation.config, shim_target, .exe) catch |err| {
+                switch (err) {
+                    error.UnsupportedTarget => {
+                        // Create a nice formatted error report
+                        const result = platform_validation.createUnsupportedTargetResult(
+                            platform_source,
+                            shim_target,
+                            .exe,
+                            validation.config,
+                        );
+                        _ = platform_validation.renderValidationError(allocs.gpa, result, stderrWriter());
+                        return error.UnsupportedTarget;
+                    },
+                    else => {},
+                }
+            };
+        } else |err| {
+            switch (err) {
+                error.MissingTargetsSection => {
+                    // Warning only - platform may still work for native builds
+                    // Don't block execution, just inform the user
+                    std.log.debug("Platform at '{s}' has no targets section", .{platform_source});
+                },
+                else => {
+                    std.log.debug("Could not validate platform header: {}", .{err});
```

Comment:
We should make these proper errors in a follow up. "targets" section is required, not just a warning. If we cannot validate the header this is also an error.